From 26bd1204d070b56eaa13f5a70bca82be0ed6158f Mon Sep 17 00:00:00 2001
From: Javier Martinez Canillas <javier.martinez@collabora.co.uk>
Date: Thu, 24 Jan 2013 09:07:53 +0100
Subject: [PATCH 049/859] af_bus: make NETFILTER_DBUS depend on NETFILTER

some users report to had a build error when NETFILTER config
option is not set:

net/bus/af_bus.c: In function 'bus_sendmsg':
net/bus/af_bus.c:1511:7: warning: 'rcp_cnt' may be used uninitialized in
this function
  CC      net/bus/garbage.o
  LD      net/bus/af-bus.o
  CC      net/bus/nfdbus/nfdbus.o
net/bus/nfdbus/nfdbus.c: In function 'nfdbus_init':
net/bus/nfdbus/nfdbus.c:406:2: error: invalid use of undefined type
'struct nf_hook_ops'
net/bus/nfdbus/nfdbus.c:407:2: error: invalid use of undefined type
'struct nf_hook_ops'
net/bus/nfdbus/nfdbus.c:408:2: error: invalid use of undefined type
'struct nf_hook_ops'
net/bus/nfdbus/nfdbus.c:409:2: error: invalid use of undefined type
'struct nf_hook_ops'
net/bus/nfdbus/nfdbus.c:410:2: error: invalid use of undefined type
'struct nf_hook_ops'
net/bus/nfdbus/nfdbus.c:411:2: error: implicit declaration of function
'nf_register_hook'
net/bus/nfdbus/nfdbus.c:425:2: error: implicit declaration of function
'nf_unregister_hook'
make[3]: *** [net/bus/nfdbus/nfdbus.o] Error 1
make[2]: *** [net/bus/nfdbus] Error 2
make[1]: *** [net/bus] Error 2
make: *** [net] Error 2

this is because NETFILTER_DBUS uses symbols only defined when NETFITER=y

Signed-off-by: Javier Martinez Canillas <javier.martinez@collabora.co.uk>
---
 net/bus/nfdbus/Kconfig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/net/bus/nfdbus/Kconfig b/net/bus/nfdbus/Kconfig
index 25699a1..d914a4e 100644
--- a/net/bus/nfdbus/Kconfig
+++ b/net/bus/nfdbus/Kconfig
@@ -3,7 +3,7 @@
 #
 config NETFILTER_DBUS
 	tristate "Netfilter D-bus (EXPERIMENTAL)"
-	depends on AF_BUS && CONNECTOR && EXPERIMENTAL
+	depends on AF_BUS && NETFILTER && CONNECTOR && EXPERIMENTAL
 	---help---
 	  If you say Y here, you will include support for a netfilter hook to
 	  parse D-Bus messages sent using the AF_BUS socket address family.
-- 
1.8.1.4

